misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/EndpointsFinder.java

```diff
@@ -69,13 +69,12 @@ public static JsonObject getEndpoints(String fileUri, WorkspaceManager workspace
             }
             Module defaultModule = module.get();
 
-            ProgramAnalyzerNodeVisitor nodeVisitor = new ProgramAnalyzerNodeVisitor();
-            nodeVisitor.setSemanticModel(semanticModel.get());
-            nodeVisitor.setFile(file);
-            if (range != null) {
-                nodeVisitor.setRange(range);
+            if (range == null) {
+                return null;
```

Comment:
Can't we return an empty json object here? returning null might cause NPEs in the future